src/qiboml/operations/expectation.py

```diff
@@ -55,8 +58,18 @@ def expectation(
         backend=backend,
         differentiation_rule=differentiation_rule,
     )
-    if isinstance(frontend, TensorflowBackend):
-        return _with_tf(**kwargs)
+
+    if differentiation_rule is not None:
+        if isinstance(qibo_backend, TensorflowBackend):
+            return _with_tf(**kwargs)
+
+        if isinstance(qibo_backend, JaxBackend):
+            return _with_jax(**kwargs)
```

Comment:
maybe the `_with_x` functions should directly be part of the backend, this way here you don't need to check which backend is used but just do `qibo_backend._with(...)`

Reply:
To be fair, it would be even nicer if I could construct a Qibo observable without any specific backend...

Reply:
We need a way to detect which is the frontend, and then build the differentiation rule on top of this. To have an agnostic observable doesn't seems to me a good idea, because it is the way we detect the frontend from the user code. 
Regarding the `_with(...)` suggestion, it could be indeed a nice idea, but then I see a problem: this function will be a third version of the `expectation` we have in the backends. Moreover,  it is a strange expectation since it wants as an argument a `differentiation_rule`, which (my opinion) makes sense only in this QML context.